### PR TITLE
[StarTransit] Machine Translation handling

### PR DESCRIPTION
**Pull request by Artuvazro**
Machine translation reference material is no longer imported by default into a TM.
Users can now choose if they want this machine translation material to be imported or not. If they choose to import it, it will be saved into an additional translation memory with a penalty of -1 so that it's not used for pre-translation and can't end up in the main translation memory when running "Update main translations memories".

### CODE REVIEW

**Comment by TinaCaputa** [comment]
Hi @Artuvazro can you please push your branch to Origin so I can also build the changes on my side?

Thanks.
Kind regards,
Florentina

**Comment by Artuvazro** [comment]
Hi @TinaCaputa just did it (I hope i did it correctly):
https://github.com/sdl/Sdl-Community/pull/348

If this is not what you were asking, please tell me and sorry for the misunderstanding.

File: StarTransit/Sdl.Community.StarTransit.Shared/Services/ProjectService.cs [hunk 1]
```diff
@@ -72,8 +81,11 @@ public MessageModel CreateProject(PackageModel package)
 					pair.StarTranslationMemoryMetadatas.Remove(item);
 				}
 
+				// Remove Machine Translation memories from pair.StarTranslationMemoryMetadatas, if the user requests them, they will be imported separately, but never in the main TM
+				pair.StarTranslationMemoryMetadatas.RemoveAll(item => Path.GetFileName(item.TargetFile).Contains("_AEXTR_MT_"));
```
[the remark this answers is not in the document]
Comment (Artuvazro) [reply StarTransit/Sdl.Community.StarTransit.Shared/Services/ProjectService.cs]: Would this alternative work?

`
pair.StarTranslationMemoryMetadatas.RemoveAll(item => Path.GetFileName(item?.TargetFile ?? "").Contains("_AEXTR_MT_"));`

File: StarTransit/Sdl.Community.StarTransit.UI/ViewModels/TranslationMemoriesViewModel.cs [hunk 2]
```diff
@@ -245,12 +284,28 @@ private void SetBtnName()
...
 
+		private void ImportMT()
+		{
+			if (IsImportMTChecked)
+			{
+				_package.MachineTransMem = new List<string>();
+
+				foreach (var filePath in SelectedItem.StarTranslationMemoryMetadatas)
+				{
+					if (Path.GetFileName(filePath.TargetFile).Contains("_AEXTR_MT_"))
```
[the remark this answers is not in the document]
Comment (Artuvazro) [reply StarTransit/Sdl.Community.StarTransit.UI/ViewModels/TranslationMemoriesViewModel.cs]: Same as above, would this alternative be good here?

`if (Path.GetFileName(filePath.TargetFile ?? "").Contains("_AEXTR_MT_"))`

**Comment by TinaCaputa** [comment]
Hi @Artuvazro, did you made the requested changes? I cannot see updates in the code. can you please perform the changes and commit them so it will be also reflected into pull request?

Thank you

File: StarTransit/Sdl.Community.StarTransit.Shared/Services/ProjectService.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TinaCaputa) [reply StarTransit/Sdl.Community.StarTransit.Shared/Services/ProjectService.cs]: It's good that way also.

File: StarTransit/Sdl.Community.StarTransit.UI/ViewModels/TranslationMemoriesViewModel.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TinaCaputa) [reply StarTransit/Sdl.Community.StarTransit.UI/ViewModels/TranslationMemoriesViewModel.cs]: it's good this alternative also

**Comment by TinaCaputa** [comment]
_Hi @TinaCaputa just did it (I hope i did it correctly):
#348

If this is not what you were asking, please tell me and sorry for the misunderstanding._

The #348 branch is not visible on the Github origin

**Comment by Artuvazro** [comment]
> Hi @Artuvazro, did you made the requested changes? I cannot see updates in the code. can you please perform the changes and commit them so it will be also reflected into pull request?
> 
> Thank you

Hi Tina, I haven't committed yet as I was waiting for your answer in my two doubts above. The other proposed changes have been applied in my local copy but I have not committed it yet. I'll do it as soon as I have time. Sorry for the delay. Thank you very much for answering those doubts!

**Comment by Artuvazro** [comment]
> > _Hi @TinaCaputa just did it (I hope i did it correctly):
> > #348
> > If this is not what you were asking, please tell me and sorry for the misunderstanding._
> 
> The #348 branch is not visible on the Github origin

I'm afraid I don't understand this issue. I thought that making a pull request of my branch would be enough. Could you please let me know how to proceed? Or where can I see instructions about this? Google has not given me any results. Sorry my lack of knowledge on this.

**Comment by TinaCaputa** [comment]
Hi @Artuvazro , I have requested above some changes in your code based on the code review I've done, like for example the changes from the below image:
image.

Please make the changes on your branch, commit the changes on your branch (in that moment, the pull request you've done will be refreshed and I will see the changes).

-If you use Source Tree tool, you can just right click on your branch and select option"Push to origin", in this way, we can also see your branch and check the changes before approving them to go to the Master branch.

**Comment by Artuvazro** [comment]
Hi Tina, changes applied and committed. Sorry for my misunderstanding. I thought the initial pull request was wrong and that I had to redo it in a different way. But you only needed me to push the changes. :) This is done already. Thank you very much for your patience.

**Comment by TinaCaputa** [comment]
Hi @Artuvazro , thanks for making the changes.
After I will finish my current task I'll test the features you've added and I will approve your request.

Kind regards,
Florentina.

**Comment by TinaCaputa** [comment]
Hi @Artuvazro, I've noticed that your branch is not on the orgin, so I cannot get it and test your changes. Can you please push it to the origin (See below image which shows you how to push a branch to origin using Source Tree)
image
Thanks

**Pull request merged**